src/context/WalletProvider/KeepKey/hooks/useKeepKeyVersions.ts

```diff
@@ -0,0 +1,83 @@
+import { isKeepKey, KeepKeyHDWallet } from '@shapeshiftoss/hdwallet-keepkey'
+import axios from 'axios'
+import { useEffect, useState } from 'react'
+import { useWallet } from 'hooks/useWallet/useWallet'
+
+interface VersionUrl {
+  version: string
+  url: string
+}
+
+export interface FirmwareDetails {
+  bootloader: VersionUrl
+  firmware: VersionUrl
+}
+
+interface FirmwareReleases {
+  latest: FirmwareDetails
+  hashes: {
+    bootloader: Record<string, string>
+    firmware: Record<string, string>
+  }
+}
+
+interface VersionStatus {
+  device: string
+  latest: string
+  updateAvailable: boolean
+}
+
+interface Versions {
+  bootloader: VersionStatus
+  firmware: VersionStatus
+}
+
+export const useKeepKeyVersions = () => {
+  const [versions, setVersions] = useState<Versions>()
+  const { state } = useWallet()
+  const { wallet } = state
+
+  const getBootloaderVersion = (keepKey: KeepKeyHDWallet, releases: FirmwareReleases): string => {
+    const hash = keepKey.features?.bootloaderHash.toString() ?? ''
```

Comment:
`wallet.features` only gets populated after calling `getFeatures()` (async). That may automatically happen in the `initialize` call but be aware that the property is asynchronous.

Reply:
Got it - I added a check and call in https://github.com/shapeshift/web/pull/1341/commits/543178a225b970f55f09b7736d44c9209f70c03d to mitigate this.